sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java

```diff
@@ -137,7 +93,36 @@ public static <T> Optional<T> extractCompasPrivate(TBaseElement baseElement, Cla
      * @throws ScdException throws when inconsistency between types
      */
     public static Optional<TCompasICDHeader> extractCompasICDHeader(TPrivate tPrivate) throws ScdException {
-        return extractCompasPrivate(tPrivate, TCompasICDHeader.class);
+        return Optional.of(tPrivate)
+                .filter(aPrivate -> COMPAS_ICDHEADER.getPrivateType().equals(aPrivate.getType()))
+                .map(TAnyContentFromOtherNamespace::getContent)
+                .map(Collection::stream)
+                .flatMap(contentElementsStream -> contentElementsStream
+                        .filter(JAXBElement.class::isInstance)
+                        .map(JAXBElement.class::cast)
+                        .filter(jaxbElement -> !jaxbElement.isNil())
+                        .map(JAXBElement::getValue)
+                        .map(TCompasICDHeader.class::cast)
+                        .reduce((t, t2) -> {
+                            //Vérification du nombre d'élements de même type dans le contenu du private => est-ce vraiment necessaire ??
+                            if (t.getClass() == t2.getClass()) {
+                                throw new ScdException("Expecting maximum 1 element of type "+COMPAS_ICDHEADER+" in private "+tPrivate.getType()+", but got more");
+                            }
+                            return t;
+                        })
+                );
```

Comment:
Duplicated code.
To bad that you remove `List<T> extractCompasPrivates(List<TPrivate> tPrivates, Class<T> compasClass)`. Maybe you could have replaced it with `Stream<T> extractCompasPrivates(List<TPrivate> tPrivates, Class<T> compasClass)`

Reply:
indeed, 
I created another method called `org.lfenergy.compas.sct.commons.scl.PrivateService#getPrivateStream`